app/services/scrape.py

```diff
@@ -218,3 +235,62 @@ def get_frequency(self, user_id: str) -> Tuple[TimeSeries, TimeSeries]:
             value=hour_of_day.tolist(),
         )
         return dow_resp, hod_resp
+
+    def get_ml_user(self, username: str):
+        user_fields = [
+            "created_at",
+            "default_profile",
+            "default_profile_image",
+            "description",
+            "favourites_count",
+            "followers_count",
+            "friends_count",
+            "listed_count",
+            "name",
+            "protected",
+            "screen_name",
+            "statuses_count",
+            "verified",
+        ]
+        user_api = self.api.get_user(screen_name=username)
+        user = {"crawled_at": datetime.utcnow()}
+        for field in user_fields:
+            user[field] = getattr(user_api, field)
+        return user
+
+    def _get_conversation(self, conversation_id: int):
+        return list(
+            tweepy.Paginator(
+                self.api_v2.search_recent_tweets,
+                query=f"conversation_id:{conversation_id}",
+                tweet_fields="referenced_tweets",
+            ).flatten()
+        )
+
+    def _get_retweets(self, tweet_id: int):
+        retweets = self.api.get_retweets(tweet_id, trim_user=True)
+        return [
+            {"id": tweet.id, "text": tweet.text, "parent_id": tweet_id}
+            for tweet in retweets
+        ]
+
+    def get_ml_tweets(self, user_id: Union[int, str]):
+        tweets = []
+        duration = timedelta(days=7)
+        last_week = datetime.utcnow() - duration
```

Comment:
We havent discussed whether we want to get all tweets or just some tweets in a period or as many tweets as possible yet right?

Reply:
We can only search tweet replies within a week with Recent Tweet Search

Reply:
What do you mean by `Recent Tweet Search`?

Reply:
Ah sorry I misread the documentation, I think it may can retrieve tweets older than 1 week.

https://developer.twitter.com/en/docs/twitter-api/tweets/search/api-reference/get-tweets-search-recent